internal/middleware/request_logging.go

```diff
@@ -0,0 +1,42 @@
+package middleware
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+)
+
+func AddRequestLogging(logs logrus.FieldLogger) mux.MiddlewareFunc {
```

Comment:
We were supposed to gradually migrate logging to slog using logrus only when there is no other option due to legacy dependencies.